src/Metric/MetricFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Metric;
+
+use Spiral\RoadRunner\Metrics;
+use Spiral\RoadRunner\MetricsInterface;
+
+class MetricFactory
+{
+    /**
+     * @var NullMetrics
+     */
+    private $nullMetrics;
+
+    /**
+     * @var Metrics
+     */
+    private $metrics;
+
+    /**
+     * @var bool
+     */
+    private $metricsEnabled;
+
+    public function __construct(NullMetrics $nullMetrics, Metrics $metrics, bool $metricsEnabled)
```

Comment:
Both services will always be instanciated, regardless of the config.

The factory can only receive the boolean config, and instantiate the right class in the `getMetricService()`.

Also RoadRunner provides an `RR_RPC` env var that contains the RPC address. We can use this here to not rely explicitly on `'%kernel.project_dir%/var/roadrunner_rpc.sock`.

It will require to parse the RPC DSN in the factory.

Reply:
done